### DISCUSSION TITLE
Running comfyUI in a controlled environment

### DISCUSSION THREAD

**Discussion by michael1maass** [Comfy-Org/ComfyUI#5919] [Q&A] [answered]
Hi, 

I tried to install and run ComfyUI under an Environment, where Windows 11 Policies prohibit running files from anywhere but "Program Files " in the user context. 
On first startup it tries to create a User_directory within the Program Path, whic,h again is also not allowed for a  plain user. 

Would it be to much to ask for a separation of configuration, Log and all other Files to be modified  during runtime and the actual Software that runs from the c:\program files\comfyUI Folder?!

can these path's eventually be configured within a config file, that is read and digested on startup? 

regards

Michael

**Comment by ltdrdata** [comment, marked as answer]
I don't know why you use install to there.
But you need to avoid any writable things from there.
To achieve that use `--user-directory`, `--extra-model-paths-config`, `--output-directory`, `--temp-directory`, `--input-directory` cli options.

**Reply by michael1maass** [reply]
I suppose, that was the Information I was looking for, that would obviously be somewhere in the docs, where it completely escaped my attention. Thank you!! Will try again and report!
